x509util/options.go

```diff
@@ -81,3 +95,94 @@ func WithTemplateFile(path string, data TemplateData) Option {
 		return fn(cr, o)
 	}
 }
+
+func asn1Encode(str string) string {
+	b64, err := mustASN1Encode(str)
+	if err != nil {
+		return err.Error()
+	}
+	return b64
+}
+
+func asn1Marshal(v interface{}, params ...string) string {
+	b64, err := mustASN1Marshal(v, params...)
+	if err != nil {
+		return err.Error()
+	}
+	return b64
+}
+
+func asn1Sequence(b64enc ...string) string {
+	b64, err := mustASN1Sequence(b64enc...)
+	if err != nil {
+		return err.Error()
+	}
+	return b64
+}
+
+func asn1Set(b64enc ...string) string {
+	b64, err := mustASN1Set(b64enc...)
+	if err != nil {
+		return err.Error()
+	}
+	return b64
+}
+
+func mustASN1Encode(str string) (string, error) {
```

Comment:
The naming scheme is a bit surprising considering in that in other places where a `mustX` is used (usually in tests), no `error` is returned. I guess that's because when there's an error returned, it will be reported when the template is evaluated?

So in the case of `asn1Set`, an eventual error ends up as the value that's returned and thus possibly set in a template? It could lead to surprising results, I think. I wonder if we should just keep it strict, and only have it return an error in those cases (obviating the need for the `mustX`). Or maybe have a way for reporting it, but not via the template itself, but logged or otherwise returned?

Reply:
Yes, the template interpreter fails when a function returns an error. Spring also uses `mustX` for this kind of function. But yes, it sounds good to me to fail always and remove the mustX methods.

Reply:
Fixed with 079c171